celer/homotopy.py

```diff
@@ -182,15 +183,22 @@ def celer_path(X, y, pb, eps=1e-3, n_alphas=100, alphas=None,
 
     X_dense, X_data, X_indices, X_indptr = _sparse_and_dense(X)
 
-    if weights is None:
+    if weights is None and pb in (LASSO, LOGREG):
```

Comment:
I will implement also a ``_check_pb``.
I don't think it is a good idea to implement the same thing for ``alphas``. Indeed, the ``_check_alphas`` function would have a lot of arguments.

Reply:
It's generally a good idea to keep the PR focussed on a single problem, here adding the weights to the group lasso. Other modifications belong to other PR.
Otherwise you end up with huge PRs which have less chances of being merged.
So no check_pb in this PR

Reply:
Got you! 
However, I have already implemented it. Let's make an exception only this time

Reply:
No, let's not. It does not belong to this PR. It needs it's separate discussion and I am not even sure it's a good idea. It will make reviewing and discussing longer. In fact, it already is.